gsy-matching-engine/src/connectors/substrate_connector.rs

```diff
@@ -0,0 +1,193 @@
+                    };
+                    let bid_offer_matches = matching_data.pay_as_bid();
+                    matches_clone_one.lock().unwrap().extend(bid_offer_matches);
+                    info!("Matches - {:?}", matches_clone_one.lock().unwrap());
+                } else {
+                    info!("No open orders to match");
+                }
+            })
+            .await
+            {
+                error!("Error while fetching the orderbook - {:?}", error);
+            }
+
+            if matches_clone_two.lock().unwrap().len() > 0 {
+                settle_matched_orders(node_url_clone, matches_clone_two).await;
+            }
+        }
+    }
+    error!("Subscription dropped.");
+    loop {
+        info!("Trying to reconnect...");
+        let two_seconds = time::Duration::from_millis(2000);
+        thread::sleep(two_seconds);
+        let orderbook_url = orderbook_url.lock().unwrap().to_string();
+        let node_url = node_url.lock().unwrap().to_string();
+        if let Err(error) = substrate_subscribe(orderbook_url, node_url.clone()).await {
+            error!("Error - {:?}", error);
+        }
+    }
+}
+
+async fn fetch_open_orders_from_orderbook_service(
+    url: String,
+) -> Result<(Vec<Bid>, Vec<Offer>), Error> {
+    let res = reqwest::get(url).await?;
+    info!("Response: {:?} {}", res.version(), res.status());
+    info!("Headers: {:#?}\n", res.headers());
+
+    let body = res.json::<Vec<OrderSchema>>().await?;
+    let open_bid: Vec<Bid> = body
+        .clone()
+        .into_iter()
+        .filter(|order| {
+            order.status == OrderStatus::Open && matches!(order.order, Order::Bid { .. })
+        })
+        .map(|order| order.into())
+        .collect();
+    let open_offer: Vec<Offer> = body
+        .into_iter()
+        .filter(|order| {
+            order.status == OrderStatus::Open && matches!(order.order, Order::Offer { .. })
+        })
+        .map(|order| order.into())
+        .collect();
+    Ok((open_bid, open_offer))
+}
+
+async fn send_settle_trades_extrinsic(
+    url: String,
+    matches: Vec<OtherBidOfferMatch<AccountId32>>,
+) -> Result<(), Error> {
+    let signer = PairSigner::new(AccountKeyring::Alice.pair());
+    info!("Signer: {:?}", signer.account_id());
+    let api = ClientBuilder::new()
+        .set_url(url)
+        .build()
+        .await?
+        .to_runtime_api::<gsy_node::RuntimeApi<
+            DefaultConfig,
+            PolkadotExtrinsicParams<DefaultConfig>,
+        >>();
+
+    let order_transfer = api.tx().trades_settlement().settle_trades(matches)?;
+
+    let order_submit_and_watch = order_transfer
+        .sign_and_submit_then_watch_default(&signer)
+        .await?
+        .wait_for_finalized_success()
+        .await?;
+
+    let transfer_event =
+        order_submit_and_watch.find_first::<gsy_node::trades_settlement::events::TradeCleared>()?;
+
+    if let Some(event) = transfer_event {
```

Comment:
I am really having trouble understanding this. I found https://stackoverflow.com/questions/72081657/rust-lang-what-is-if-let-somex-x-doing but i can not translate it to this case here.
I do not see where `event` is assigned or declared. Can you please help me?

Reply:
This is how `if let` works in Rust https://doc.rust-lang.org/book/ch06-03-if-let.html. Essentially it declares the `event` variable that is populated if the `transfer_event` is `Some` value and not an error.